hpx/parallel/algorithms/fill.hpp

```diff
@@ -65,10 +62,10 @@ namespace hpx { namespace parallel { inline namespace v1
               : fill::algorithm("fill")
             {}
 
-            template <typename ExPolicy, typename InIter, typename T>
+            template <typename ExPolicy, typename FwdIter, typename T>
             HPX_HOST_DEVICE
             static hpx::util::unused_type
-            sequential(ExPolicy, InIter first, InIter last,
+            sequential(ExPolicy, FwdIter first, FwdIter last,
```

Comment:
Apart from adapting the parallel algorithm interface to the standard, this sequential function requires just InputIterator. So I think that ```InIter``` is correct in this position. How do you think?
This comment is applied to many spots in this PR.